iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,217 @@
+(`request.getRequestHeader().setSequenceId()`).
+The request is transmitted to the server via the `send()` API.
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [[send request]] -->
+```cpp
+    client.loan()
+        .and_then([&](auto& request) {
+            auto guardedCtx = ctx.getScopeGuard();
+            request.getRequestHeader().setSequenceId(guardedCtx->requestSequenceId);
+            guardedCtx->expectedResponseSequenceId = guardedCtx->requestSequenceId;
+            guardedCtx->requestSequenceId += 1;
+            request->augend = guardedCtx->fibonacciLast;
+            request->addend = guardedCtx->fibonacciCurrent;
+            std::cout << APP_NAME << " Send Request: " << guardedCtx->fibonacciLast << " + "
+                      << guardedCtx->fibonacciCurrent << std::endl;
+            request.send();
+        })
+        .or_else([](auto& error) {
+            std::cout << "Could not allocate Request! Return value = " << static_cast<uint64_t>(error) << std::endl;
+        });
+```
+
+In the same while loop the client receives the responses from the server using `take()`
+and extract the sequence id with `response.getResponseHeader().getSequenceId()`.
+When the server response comes in the correct order, the received sum is re-used to
+insert it to the `addend` of the next request to send.
+
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [[take response]] -->
+```cpp
+    while (client.take().and_then([&](const auto& response) {
+        auto receivedSequenceId = response.getResponseHeader().getSequenceId();
+        if (receivedSequenceId == guardedCtx->expectedResponseSequenceId)
+        {
+            guardedCtx->fibonacciLast = guardedCtx->fibonacciCurrent;
+            guardedCtx->fibonacciCurrent = response->sum;
+            std::cout << APP_NAME << " Got Response : " << guardedCtx->fibonacciCurrent << std::endl;
+        }
+        else
+        {
+            std::cout << "Got Response with outdated sequence ID! Expected = "
+                      << guardedCtx->expectedResponseSequenceId << "; Actual = " << receivedSequenceId
+                      << "! -> skip" << std::endl;
+        }
+    }))
+    {
+    }
+```
+
+### Server using Listener
+
+At first, the includes for the server port and request-response types and runtime are needed.
+<!-- [geoffrey] [geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp] [iceoryx includes] -->
+```cpp
+#include "request_and_response_types.hpp"
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/notification_callback.hpp"
+#include "iceoryx_posh/popo/server.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+First, a callback is created that shall be called when server receives a response.
+In this case the calculation is here done too but more resource consuming tasks
```

Comment:
Hard to understand. Maybe you mean 
In this case the calculation and the sending of the response is done in the listener callback. If there are more resource consuming tasks, this could also be outsourced with a thread pool to handle the requests